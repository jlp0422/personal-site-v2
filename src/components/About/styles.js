import styled from '@emotion/styled'

export const Copy = styled.p`
  font-size: 2rem;
  padding: 0.5rem 0;
`

export const ContactList = styled.div`
  > div {
    font-family: ${({ theme }) => theme.fonts.primary};
    font-size: 1.8rem;
  }
  span,
  a {
    font-size: 1.8rem;
  }
  a {
    cursor: pointer;
  }
`

export const ContentWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 350px;
  grid-column-gap: 3%;
  @media only screen and (max-width: 800px) {
    grid-template-columns: 1fr 250px;
  }
  @media only screen and (max-width: 640px) {
    display: block;
  }
`

export const ImageWrapper = styled.div`
  @media only screen and (max-width: 640px) {
    width: 300px;
    margin: 2rem auto;
  }
`
