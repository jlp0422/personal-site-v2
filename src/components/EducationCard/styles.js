import styled from '@emotion/styled'
import {
  CardFlipBack,
  CardFlipContainer,
  CardFlipInner,
  CardShared,
  ExpInfo,
  ExpTitle
} from '../shared'

export const Container = styled(CardFlipContainer)`
  text-align: center;
  @media only screen and (max-width: 640px) {
    height: 250px;
  }
`

export const InnerContainer = styled(CardFlipInner)`
  box-shadow: ${({ theme }) => theme.palette.boxShadow.dark};
`

export const CardFront = styled(CardShared)`
  background-color: ${({ theme }) => theme.palette.black.darkest};
`

export const CardBack = styled(CardFlipBack)`
  background-color: ${({ theme }) => theme.palette.white.lighter};
`

export const Title = ExpTitle

export const Info = styled(ExpInfo)`
  width: 90%;
  grid-template-columns: repeat(2, 1fr);
`
