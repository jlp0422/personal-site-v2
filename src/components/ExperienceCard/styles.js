import styled from '@emotion/styled'
import {
  ExpInfo,
  ExpTitle,
  CardFlipContainer,
  CardFlipInner,
  CardFlipBack,
  CardShared
} from '../shared'

export const Container = styled(CardFlipContainer)`
  display: grid;
  text-align: center;
  margin: 2.5rem auto;
  @media only screen and (max-width: 640px) {
    height: 250px;
  }
`

export const InnerContainer = styled(CardFlipInner)`
  box-shadow: ${({ theme }) => theme.palette.boxShadow.light}
`

export const CardFront = styled(CardShared)`
  padding: 1.5rem;
  background-color: ${({ theme }) => theme.palette.black.darkest};
`

export const CardBack = styled(CardFlipBack)`
  padding: 1.5rem;
  background-color: ${({ theme }) => theme.palette.white.lighter}
`

export const Title = ExpTitle

export const Info = styled(ExpInfo)`
  width: 100%;
  align-items: center;
  @media only screen and (max-width: 640px) {
    grid-template-columns: 1fr;
  }
`

export const LogoContainer = styled.div`
  display: block;
  visibility: visible;
  width: 60px;
  margin: 1rem auto 0;
  @media only screen and (max-width: 640px) {
    display: none;
    visibility: hidden;
  }
`
