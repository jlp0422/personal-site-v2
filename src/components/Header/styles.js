import styled from '@emotion/styled'
import { Link } from 'gatsby'
import { ActiveLink } from '../../components/shared/Styled'

export const StyledHeader = styled.header`
  background: ${({ theme }) => theme.palette.link.secondary};
  margin-bottom: 1.5rem;
`

export const Div = styled.div`
  margin: 0 auto;
  max-width: 1200px;
  padding: 2rem 1.2rem;
  display: grid;
  grid-template-columns: 120px 120px 1fr 120px 120px;
  justify-items: center;
  align-items: center;
  grid-column-gap: 2rem;
  @media only screen and (max-width: 640px) {
    padding: 0.8rem;
    display: flex;
    align-items: center;
    justify-content: center;
  }
`

export const H1 = styled.h1`
  margin: 0;
  text-align: center;
  text-transform: uppercase;
`

export const HeaderLink = styled(Link)`
  letter-spacing: 0.3rem;
  font-family: ${({ theme }) => theme.fonts.header};
  color: white;
  text-decoration: none;
  font-size: 4.5rem;
  @media only screen and (max-width: 800px) {
    visibility: hidden;
    display: none;
  }
`

export const LogoLink = styled(Link)`
  margin: 0;
  visibility: hidden;
  display: none;
  @media only screen and (min-width: 640px) and (max-width: 800px) {
    visibility: visible;
    display: block;
  }
`

export const NavLink = styled(ActiveLink)`
  @media only screen and (max-width: 640px) {
    display: none;
    visibility: hidden;
  }
`

export const MobileLogo = styled.button`
  margin: 0;
  visibility: hidden;
  display: none;
  @media only screen and (max-width: 639px) {
    visibility: visible;
    display: block;
  }
`

export const List = styled.ul`
  padding: 0 0 1rem;
  list-style: none;
  margin-top: 0;
  visibility: hidden;
  display: none;
  @media only screen and (max-width: 640px) {
    visibility: visible;
    display: block;
  }
`
