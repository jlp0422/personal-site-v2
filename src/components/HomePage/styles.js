import { keyframes } from '@emotion/core'
import styled from '@emotion/styled'
import { Link } from 'gatsby'

export const spin = keyframes`
  from {
    transform: rotate(0deg)
  }

  to {
    transform: rotate(360deg)
  }
`

export const GridContainer = styled.div`
  display: grid;
  grid-template: repeat(2, 1fr) / repeat(2, 1fr);
  grid-gap: 2rem 2rem;
  justify-content: center;
  align-items: center;
  text-align: center;
  margin-top: 1rem;
  > * {
    margin: 0;
  }
  @media only screen and (max-width: 800px) {
    grid-template: repeat(4, 1fr) / 1fr;
  }
`

export const LinkBox = styled(Link)`
  background-color: ${({ color }) => color};
  padding: 8rem;
  font-family: ${({ theme }) => theme.fonts.nav};
  text-decoration: none;
  transition: all 0.2s linear;
  > h3 {
    margin: 1.5rem;
  }
  @media only screen and (min-width: 641px) {
    :hover {
      transform: scale(1.05);
      div {
        animation: ${spin} 2s linear infinite;
      }
    }
  }
  @media only screen and (max-width: 800px) {
    padding: 5rem;
  }
  @media only screen and (max-width: 640px) {
    padding: 1rem;
    > h3 {
      margin: 0.5rem;
    }
  }
`
