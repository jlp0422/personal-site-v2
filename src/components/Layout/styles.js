import styled from '@emotion/styled'

export const ContainerDiv = styled.div`
  margin: 0 auto;
  max-width: 1200px;
  padding: 0 1.5rem 1.5rem;
  min-height: 85vh;
  display: grid;
  grid-template-rows: 1fr auto;
`
