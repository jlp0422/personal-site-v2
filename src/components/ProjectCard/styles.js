import styled from '@emotion/styled'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: white;
  margin: 1rem 0;
  padding: 1.5rem;
  border: 1px solid black;
  transition: all 0.2s linear;
  --webkit-transition: all 0.2s linear;
  :hover {
    box-shadow: ${({ theme }) => theme.palette.boxShadow.primary};
    transform: translate(-8px, -8px);
    --webkit-transform: translate(-8px, -8px);
  }
  @media only screen and (max-width: 480px) {
    box-shadow: ${({ theme }) => theme.palette.boxShadow.primary};
    :hover {
      transform: none;
    }
  }
`

export const LinkContainer = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  * + * {
    margin-top: 0;
  }
`
