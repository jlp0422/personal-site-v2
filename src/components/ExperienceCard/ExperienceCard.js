import React from 'react'
import { DataItem } from '../../components/shared'
import { joinString } from '../../helpers'
import Image from '../Image'
import { Container, Info, LogoContainer, Title } from './styles'

const ExperienceCard = ({ experience }) => {
  const { title, location, company, startDate, endDate } = experience
  const normalizedCompany = joinString(company)
  const dataProps = {
    size: 'medium',
    align: 'center',
    background: 'dark'
  }
  return (
    <Container>
      <Title>{title}</Title>
      <Info>
        <DataItem value={location} label="Location" {...dataProps} />
        <DataItem value={company} label="Company" {...dataProps} />
        <DataItem
          value={`${startDate} - ${endDate}`}
          label="Dates"
          {...dataProps}
        />
      </Info>
      <LogoContainer>
        <Image
          queryKey={`${normalizedCompany}Logo`}
          style={{ width: '60px' }}
        />
      </LogoContainer>
    </Container>
  )
}

export default ExperienceCard
