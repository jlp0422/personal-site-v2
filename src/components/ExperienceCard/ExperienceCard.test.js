import React from 'react'
import renderer from 'react-test-renderer'
import ExperienceCard from './ExperienceCard'
import { ThemeWrapper, DataItem } from '../shared'
import { Title, Info, CardBack } from './styles'

describe('ExperienceCard', () => {
  const getOutput = () =>
    renderer.create(
      <ThemeWrapper>
        <ExperienceCard
          experience={{
            title: 'My first job',
            location: 'Los Angeles, CA',
            company: 'ABC',
            startDate: 'Jan 1900',
            endDate: 'Jan 2000',
            details: ['it was fun', 'it was great', 'it was awesome']
          }}
        />
      </ThemeWrapper>
    )

  it('Renders correctly', () => {
    const tree = getOutput().toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Displays the title in all uppercase', () => {
    const instance = getOutput().root
    expect(instance.findByType(Title).props.children).toBe('MY FIRST JOB')
  })

  it('Displays the Location value and label', () => {
    const instance = getOutput().root
    const locationProps = instance.findAllByType(DataItem)[0].props
    expect(locationProps.value).toBe('Los Angeles, CA')
    expect(locationProps.label).toBe('Location')
    expect(locationProps.align).toBe('center')
    expect(locationProps.size).toBe('medium')
    expect(locationProps.background).toBe('dark')
  })

  it('Displays the Company value and label', () => {
    const instance = getOutput().root
    const companyProps = instance.findAllByType(DataItem)[1].props
    expect(companyProps.value).toBe('ABC')
    expect(companyProps.label).toBe('Company')
    expect(companyProps.align).toBe('center')
    expect(companyProps.size).toBe('medium')
    expect(companyProps.background).toBe('dark')
  })

  it('Displays the Dates value and label', () => {
    const instance = getOutput().root
    const dateProps = instance.findAllByType(DataItem)[2].props
    expect(dateProps.value).toBe('Jan 1900 - Jan 2000')
    expect(dateProps.label).toBe('Dates')
    expect(dateProps.align).toBe('center')
    expect(dateProps.size).toBe('medium')
    expect(dateProps.background).toBe('dark')
  })

  it('Displays the array of details', () => {
    const instance = getOutput().root
    const getChildAtIndex = index =>
      instance.findByType(CardBack).props.children[index].props
        .dangerouslySetInnerHTML.__html

    expect(instance.findByType(CardBack).props.children.length).toBe(3)
    expect(getChildAtIndex(0)).toBe('<p>it was fun.</p>')
    expect(getChildAtIndex(1)).toBe('<p>it was great.</p>')
    expect(getChildAtIndex(2)).toBe('<p>it was awesome.</p>')
  })
})
