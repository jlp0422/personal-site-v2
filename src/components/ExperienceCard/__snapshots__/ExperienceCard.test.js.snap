// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceCard Renders correctly 1`] = `
<div
  className="css-1yyoknq"
>
  <div
    className="css-1vqdcjn"
  >
    <div
      className="css-h5v37p"
    >
      <h3
        className="css-1b8m15i"
      >
        MY FIRST JOB
      </h3>
      <div
        className="css-1fflyas"
      >
        <div
          className="css-gmm171"
          size="medium"
        >
          <p>
            Los Angeles, CA
          </p>
          <p>
            Location
          </p>
        </div>
        <div
          className="css-gmm171"
          size="medium"
        >
          <p>
            ABC
          </p>
          <p>
            Company
          </p>
        </div>
        <div
          className="css-gmm171"
          size="medium"
        >
          <p>
            Jan 1900 - Jan 2000
          </p>
          <p>
            Dates
          </p>
        </div>
      </div>
    </div>
    <div
      className="css-gohb89"
    >
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>it was fun.</p>",
          }
        }
      />
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>it was great.</p>",
          }
        }
      />
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>it was awesome.</p>",
          }
        }
      />
    </div>
  </div>
</div>
`;
